src/virtual_renderer.js

```diff
@@ -1620,6 +1620,24 @@ class VirtualRenderer {
                 className: "ace_ghost_text"
             };
             this.session.widgetManager.addLineWidget(this.$ghostTextWidget);
+
+            // Check wether the line widget fits in the part of the screen currently in view
+            var pixelPosition = this.$cursorLayer.getPixelPosition(insertPosition, true);
+            var el = this.container;
+            var height = el.clientHeight;
+            var fitsY = textLines.length * this.lineHeight < height - pixelPosition.top;
+
+            // If it fits, no action needed
+            if (fitsY) return;
+            
+            // If it can fully fit in the screen, scroll down until it fits on the screen
+            // if it cannot fully fit, scroll so that the cursor is at the top of the screen
+            // to fit as much as possible.
+            if (textLines.length * this.lineHeight < height) {
```

Comment:
**nitpick:** we multiply `textLines.length * this.lineHeight` twice, so we can store the value and reuse it across lines 1628 and 1636.

Reply:
changed this